partner_event/models/event_registration.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# © 2014 Tecnativa S.L. - Pedro M. Baeza
+# © 2015 Tecnativa S.L. - Javier Iniesta
+# © 2016 Tecnativa S.L. - Antonio Espinosa
+# © 2016 Tecnativa S.L. - Vicent Cubells
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, fields
+
+
+class EventRegistration(models.Model):
+    _inherit = "event.registration"
+
+    def _prepare_partner(self, vals):
+        return {
+            'name': vals.get('name') or vals.get('email'),
+            'email': vals.get('email', False),
+            'phone': vals.get('phone', False),
+        }
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('partner_id') and vals.get('email'):
+            partner_model = self.env['res.partner']
+            event_model = self.env['event.event']
+            partner_id = False
+            # Look for a partner with that email
+            email = vals.get('email').replace('%', '').replace('_', '\\_')
+            partners = partner_model.search(
+                [('email', '=ilike', email)])
+            event = event_model.browse(vals['event_id'])
+            if partners:
+                partner_id = partners[0].id
+                vals['name'] = vals.get('name') or partners[0].name
+                vals['phone'] = vals.get('phone') or partners[0].phone
+            elif event.create_partner:
+                # Create partner
+                partner = partner_model.sudo().create(
+                    self._prepare_partner(vals))
+                partner_id = partner.id
+            vals['partner_id'] = partner_id
+        return super(EventRegistration, self).create(vals)
+
+    @api.multi
+    def partner_data_update(self, data):
+        reg_fields = ['name', 'email', 'phone']
```

Comment:
Normally if you need to use a list only for checking existence with the `in` operator, it is better to use sets. Not very critical in a list of 3, so not blocking, but a good point to have in mind in general :wink:.